CONTRIBUTING.md

```diff
@@ -95,6 +96,90 @@ have more control over the process and debug any issues.
+    | extend BuildLink = tostring(parse_json(Data)._links.web.href)
+    | project StartTime, BuildId, SourceBranch, SourceVersion, PipelineName, PullRequestLink, BuildLink
+    | join kind=inner
+      (
+      BuildLogLine
+      | where Timestamp > beginTime
+      | where LogId == lintDiffLogId
+      | where Message startswith_cs lintDiffLogMessageStart
+      | extend LintDiffViolations = parse_json(Message)
+      | mv-expand LintDiffViolations
+      | extend ViolationCode = tostring(LintDiffViolations.code)
+      | where ViolationCode == ruleName
+      )
+      on BuildId
+    | summarize count() by Time=bin(Timestamp, violationTimeBin), ViolationCode, PullRequestLink, BuildLink
+    | sort by count_ desc
+    ```
+    1. Wait until your rule has run on at least (**TODO**) different PRs and you have verified there are no false positives
+    1. It might also be helpful to view more than only the LintDiff results. You can use this Kusto query to see violations
+    of your new rule for an API spec without taking into account the previous version of the spec:
+    ```kusto
+    // edit these as desired
+    let beginTime = ago(7d);
+    let violationTimeBin = 10m;
+    let includeStagingPipeline = true;
+    let includeProdPipeline = false;
+    let ruleName = "ReservedResourceNamesModelAsEnum";
+    // do not edit these
+    let prodPipelineName = "Azure.azure-rest-api-specs-pipeline";
+    let stagingPipelineName = "Azure.azure-rest-api-specs-pipeline-staging";
+    let lintDiffLogId = 62;
+    let lintDiffLogMessageStart = "          \"code\":";
+    Build
+    | where StartTime > beginTime
+    | extend PipelineName = parse_json(Data).definition.name
+    | where PipelineName in (iff(includeStagingPipeline, stagingPipelineName, prodPipelineName), iff(includeProdPipeline, prodPipelineName, stagingPipelineName))
+    | extend SplitBranch = split(SourceBranch, "/")
+    | extend PullRequestLink = strcat("https://github.com/", SplitBranch[3], "/", SplitBranch[4], "/pull/", SplitBranch[5])
+    | extend BuildLink = tostring(parse_json(Data)._links.web.href)
+    | project StartTime, BuildId, SourceBranch, SourceVersion, PipelineName, PullRequestLink, BuildLink
+    | join kind=inner 
+      (
+      BuildLogLine
+      | where Timestamp > beginTime
+      | where LogId == lintDiffLogId
+      | where Message startswith_cs lintDiffLogMessageStart
+      | extend ViolationCode=trim_end('",', substring(Message, indexof(Message, ': "')+3))
+      | where ViolationCode == ruleName
+      )
+      on BuildId
+    | summarize count() by Time=bin(Timestamp, violationTimeBin), ViolationCode, PullRequestLink, BuildLink
+    | sort by count_ desc 
+    ```
+    where the rule was incorrectly marked as violated.
+1. Once you verify the rules work correctly, roll them out to the production pipeline following the process defined in
```

Comment:
Is it just about reverting? A new package needs to be created and deployed right? Can we link the e2e steps here?

Reply:
This - "Deploy to Prod LintDiff"

Reply:
Yes, sorry, I wasn't clear. One has to deploy to prod and also revert the process I mentioned. I.e. remove the `stagingOnly: true`.

Reply:
Actually mentioning the deploy to prod is redundant because the section about setting rules to run only in staging already mentions them:

- https://github.com/Azure/azure-openapi-validator/blob/cd28c4aa518a2c8e0aa64267cac324a31e9ce839/CONTRIBUTING.md#how-to-set-a-spectral-rule-to-run-only-in-staging

Reply:
I think we can simplify this by requiring all new rules to be set to staging-only. Then, when we do a release we remove the staging-only property for each rule we want to include or add the property back if we want to remove the rule.